components/motor/gpiostepper/gpiostepper.go

```diff
@@ -85,62 +85,58 @@ func (cfg *Config) Validate(path string) ([]string, error) {
 
 func init() {
 	resource.RegisterComponent(motor.API, model, resource.Registration[motor.Motor, *Config]{
-		Constructor: func(
-			ctx context.Context,
-			deps resource.Dependencies,
-			conf resource.Config,
-			logger logging.Logger,
-		) (motor.Motor, error) {
-			actualBoard, motorConfig, err := getBoardFromRobotConfig(deps, conf)
-			if err != nil {
-				return nil, err
-			}
-
-			return newGPIOStepper(ctx, actualBoard, *motorConfig, conf.ResourceName(), logger)
-		},
-	})
+		Constructor: newGPIOStepper,
+	},
+	)
 }
 
-func getBoardFromRobotConfig(deps resource.Dependencies, conf resource.Config) (board.Board, *Config, error) {
-	motorConfig, err := resource.NativeConfig[*Config](conf)
-	if err != nil {
-		return nil, nil, err
-	}
-	if motorConfig.BoardName == "" {
-		return nil, nil, errors.New("expected board name in config for motor")
+
+// TODO (rh) refactor this driver so that the enable and direction logic is at the beginning of each API call
+// and the step -> position logic is the only thing being handled by the background thread.
+// right now too many things can be called out of lock, this function is only called from the constructor, CLose
+// the doCycle step routine, and should not be called elsewhere since there's no lock in in to ptoect the enable pins
+func (m *gpioStepper) enable(ctx context.Context, high bool) error {
+	var err error
+	if m.enablePinHigh != nil {
+		err = multierr.Combine(err, m.enablePinHigh.Set(ctx, high, nil))
 	}
-	b, err := board.FromDependencies(deps, motorConfig.BoardName)
-	if err != nil {
-		return nil, nil, err
+
+	if m.enablePinLow != nil {
+		err = multierr.Combine(err, m.enablePinLow.Set(ctx, !high, nil))
 	}
-	return b, motorConfig, nil
+
+	return err
```

Comment:
Moved enable up with no changes.
Added a long warning not to use this outside of the constructor, Close and doCycle because we cowboy it and don't lock the pins stored on the motor struct.